clevertap-core/src/main/java/com/clevertap/android/sdk/CallbackManager.java

```diff
@@ -53,8 +53,7 @@ public class CallbackManager extends BaseCallbackManager {
 
     private NotificationRenderedListener notificationRenderedListener;
 
-    //TODO this list could be a synchronized set
-    private final List<OnInitCleverTapIDListener> onInitCleverTapIDListeners = new ArrayList<>();
+    private final List<OnInitCleverTapIDListener> onInitCleverTapIDListeners =  Collections.synchronizedList(new ArrayList<>());
```

Comment:
@piyush-kukadiya where is the code where we're adding callback objects to this synchronized list?

Reply:
here it is https://github.com/CleverTap/clevertap-android-sdk/blob/master/clevertap-core/src/main/java/com/clevertap/android/sdk/CleverTapAPI.java#L3255

Reply:
Understood.